pnet-idp-client-saml2/src/test/java/at/porscheinformatik/idp/saml2/Saml2ResponseProcessorTest.java

```diff
@@ -478,13 +491,49 @@ protected TokenAndResponse buildTokenAndResponse(boolean signed, boolean invalid
         storeSessionAge(request, Optional.ofNullable(sessionAge));
 
         request.addParameter("SAMLResponse", base64Response);
+        request.addParameter(RELAY_STATE_PARAM, responseRelayState);
 
-        Saml2AuthenticationToken token = new Saml2AuthenticationToken(buildRelyingPartyRegistration(), stringResponse);
+        RelyingPartyRegistration relyingPartyRegistration = buildRelyingPartyRegistration();
+        AbstractSaml2AuthenticationRequest authenticationRequest = Saml2PostAuthenticationRequest
+            .withRelyingPartyRegistration(relyingPartyRegistration)
+            .samlRequest("dummy request")
+            .relayState(requestedRelayState)
+            .build();
+        Saml2AuthenticationToken token = new Saml2AuthenticationToken(relyingPartyRegistration, stringResponse, authenticationRequest);
         token.setDetails(new HttpRequestContext(request));
 
         return new TokenAndResponse(token, response);
     }
 
+    @Test
+    public void failsOnMissingResponseRelayState() throws Exception
```

Comment:
Maybe you can move the tests befor the `buildTokenAndResponse` methods. All other tests are also before the methods.

Reply:
Just moved the tests right next to other test methods.